### PR TITLE
feat: Record HTTP server request and response headers

### PR DESCRIPTION
**Pull request by dustinbyrne**

### CODE REVIEW

File: src/main/java/com/appland/appmap/output/v1/Event.java [hunk 1]
```diff
@@ -363,10 +365,11 @@ public Event setHttpServerResponse(Integer status) {
    * @return {@code this}
    * @see <a href="https://github.com/applandinc/appmap#http-server-response-attributes">GitHub: AppMap - HTTP server response attributes</a>
    */
-  public Event setHttpServerResponse(Integer status, String mimeType) {
+  public Event setHttpServerResponse(Integer status, String mimeType, Map<String, String> headers) {
     this.httpServerResponse = new HttpServerResponse()
         .setStatus(status)
-        .setMimeType(mimeType);
+        .setMimeType(mimeType)
```
Comment (kgilpin) [inline src/main/java/com/appland/appmap/output/v1/Event.java:371]: Mime type has been dropped from the AppMap spec as a separate field.

**Comment by kgilpin** [comment]
Looking good so far - mime type can be dropped.

File: src/main/java/com/appland/appmap/output/v1/HttpServerResponse.java [hunk 2]
```diff
@@ -14,6 +16,9 @@
   @JSONField(name = "mime_type")
   public String mimeType;
```
Comment (kgilpin) [inline src/main/java/com/appland/appmap/output/v1/HttpServerResponse.java:17]: This can be dropped.

**Comment by apotterri** [comment]
**Please don't merge** , the fixup commit needs to be squashed.

These changes get rid of the MIME type on the server response, as well as adding a couple of simple tests. I also tweaked the petclinic test script to make it more useful for running interactively.

**Comment by dustinbyrne** [comment]
Looks good to me 👍

**Pull request merged**